scripts/Jenkinsfile

```diff
@@ -1,43 +1,58 @@
 pipeline 
```

Comment:
Can we make this a separate CI pipeline? that runs only when `pyproject.toml` is touched upon, or runs only on release branches. The current pipeline by itself takes a lot of time to run, adding this for every PR and every push is not a good idea.

Reply:
We will be pulling nightly docker which will pull only once every day. Let's wait for the CI time with this pipeline to make a decision on this.

Reply:
Agree, docker essentially checks for creation of venv only, and does basic check of installation success, something like `python -c "import QEfficient"`, we should keep it specific to changes in packages only for now. (i.e trigger only when packages are changed)

Reply:
Update- 
1. I've successfully implemented a nightly build pipeline that retrieves the latest Docker image from the registry at midnight. This ensures that we're always working with the most up-to-date image.

2. Leveraging this setup, we can seamlessly run tests throughout the day by creating Docker containers. 

3. Additionally, any modifications to the "pyproject.toml" file will trigger a new pipeline. This pipeline will meticulously build the Docker image from scratch and execute tests within it. Our testing process remains robust and aligned with the latest code changes. (Under progress)